doc/man3/EVP_EncryptInit.pod

```diff
@@ -156,6 +160,16 @@ EVP_CIPHER_CTX_mode
  int EVP_Cipher(EVP_CIPHER_CTX *ctx, unsigned char *out,
                 const unsigned char *in, unsigned int inl);
 
+ int EVP_CIPHER_can_pipeline(const EVP_CIPHER *cipher, int enc);
+ int EVP_CipherPipelineInit(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
+                                  const unsigned char *key, size_t numpipes,
+                                  const unsigned char **iv, int enc);
```

Comment:
I don't see the benefit of doing this.

Reply:
There is a huge benefit - there is no point in having any special handling of the iv and key length in libcrypto.

Also splitting the encrypt and decrypt init functions allows to avoid splitting the calls in libcrypto as well.

We should not repeat the mistakes we did in the original EVP_Cipher* APIs.

Reply:
Done